src/util/mod.rs

```diff
@@ -149,6 +150,88 @@ impl VersionTriple {
     }
 }
 
+#[derive(Debug, Error)]
+pub enum VersionDoubleError {
+    #[error("Failed to parse major version from {version:?}: {source}")]
+    MajorInvalid {
+        version: String,
+        source: std::num::ParseIntError,
+    },
+    #[error("Failed to parse minor version from {version:?}: {source}")]
+    MinorInvalid {
+        version: String,
+        source: std::num::ParseIntError,
+    },
+    #[error(
+        "Failed to parse version string {version:?}: string must be in format <major>[.minor]"
+    )]
+    VersionStringInvalid { version: String },
+}
+
+// Generic version double
+#[derive(Clone, Copy, Debug, Eq, Ord, PartialEq, PartialOrd)]
+pub struct VersionDouble {
+    pub major: u32,
+    pub minor: u32,
+}
+
+impl Display for VersionDouble {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}.{}", self.major, self.minor)
+    }
+}
+
+impl Serialize for VersionDouble {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.collect_str(self)
+    }
+}
+
+impl VersionDouble {
+    pub const fn new(major: u32, minor: u32) -> Self {
+        Self { major, minor }
+    }
+
+    pub fn from_str(v: &str) -> Result<Self, VersionDoubleError> {
+        if !v.contains(".") {
+            return Ok(VersionDouble {
+                major: v
+                    .parse()
+                    .map_err(|source| VersionDoubleError::MajorInvalid {
+                        version: v.to_owned(),
+                        source,
+                    })?,
+                minor: 0,
+            });
+        }
+
+        if v.split(".").count() == 2 {
```

Comment:
You could match on this instead, which you can also use for the `contains` case above... I also feel like there's probably a cool way to golf all of this (though I'm having trouble thinking of one)